src/Host.cpp

```diff
@@ -802,15 +802,14 @@ bool Host::installPackage(const QString& fileName, int module )
                 pLabel->setText( tr( "Unpacking package:\n\"%1\"\nplease wait..." ).arg( packageName ) );
             }
         }
+        pUnzipDialog->hide(); // Must hide to change WindowModality
         pUnzipDialog->setWindowTitle( tr( "Unpacking" ) );
-        pUnzipDialog->hide();
         pUnzipDialog->setWindowModality( Qt::ApplicationModal );
         pUnzipDialog->show();
         qApp->processEvents();
         pUnzipDialog->raise();
-        pUnzipDialog->show(); // Must do this to ensure modality is applied
-        pUnzipDialog->update();
-        qApp->processEvents(); // Try to ensure we are on top of any other dialogs
+        pUnzipDialog->repaint(); // Force a redraw
```

Comment:
This forces a repaint of the Unpacking widget which is a mandatory redrawing of it -  I cannot see _why_ it is not getting repainted for you @vadi2, especially with the `qApp->processEvents()` call immediately afterwards which the system should be using to "refresh" anything else it thinks need repainting, etc. I'm baffled why it isn't being properly filled in for you...

The only further thing I can suggest is a:

    if( pLabel )
    {
        pLabel->repaint();
    }

before this line but I am grasping at straws here as I don't see why it isn't working for you - could you try it and see if that changes things?